server/entity/experience.go

```diff
@@ -0,0 +1,118 @@
+package entity
+
+import "math"
+
+var maxLevel int32 = math.MaxInt32
+
+type ExperienceManager struct {
+	level    int32
+	progress float64
+	totalXP  int
+}
+
+// NewExperienceManager return a ExperienceManager with default value of level, progress and totalXP that are equal to 0.
+func NewExperienceManager() *ExperienceManager {
+	return &ExperienceManager{}
+}
+
+// Level get xp level.
+func (e *ExperienceManager) Level() int32 {
+	return e.level
+}
+
+// Progress get the progress of the xp, the value are Between 0.00 and 1.00.
+func (e *ExperienceManager) Progress() float64 {
+	return e.progress
+}
+
+// MaxLevel get the max xp level.
+func (e *ExperienceManager) MaxLevel() int32 {
+	return maxLevel
+}
+
+// TotalXP get the total xp collected.
+func (e *ExperienceManager) TotalXP() int {
+	return e.totalXP
+}
+
+// GetExperienceToNextLevel get the total experience netted to level up.
+func (e *ExperienceManager) GetExperienceToNextLevel() int {
+	return e.GetExperienceForLevel(e.level)
+}
+
+// GetExperienceNettedToLevelUp get xp that you need to level up with reducing xp that you already have.
+func (e *ExperienceManager) GetExperienceNettedToLevelUp() int {
+	return e.GetExperienceToNextLevel() - e.ProgressToXp(e.progress, e.level)
+}
+
+// GetExperienceForLevel get the amount xp to level up in a specific level.
+func (e *ExperienceManager) GetExperienceForLevel(level int32) int {
```

Comment:
```suggestion
func (e *ExperienceManager) ExperienceForLevel(level int32) int {
```

Reply:
I think we should decide if we want to consistently go for `Experience` or for `XP`.